File: SoftUni Lessons/Python Development/Python Basics April 2019/Past Exams and Problems/3 - 3 and 4 November 2018/0 - My Exam/05. Series .py
budget = float(input())
count_tv_series = int(input())

total_price = 0

for serial in range(count_tv_series):
    name_serial = input()
    price_serial = float(input())
    if name_serial == "Thrones":
        total_price += (price_serial * 0.5)
    elif name_serial == "Lucifer":
        total_price += (price_serial * 0.6)
    elif name_serial == "Protector":
        total_price += (price_serial * 0.7)
    elif name_serial == "TotalDrama":
        total_price += (price_serial * 0.8)
    elif name_serial == "Area":
        total_price += (price_serial * 0.9)
    else:
        total_price += price_serial

if budget >= total_price:
    print(f"You bought all the series and left with {(budget - total_price):.2f} lv.")
else:
    print(f"You need {(total_price - budget):.2f} lv. more to buy the series!")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/21. МИНАЛИ ИЗПИТИ/04. Online Exam - 15 and 16 June 2019/05. Series.js
function series(input) {
    // Mask
    let discounted = {
        'Thrones': 0.5,
        'Lucifer': 0.6,	
        'Protector': 0.7,
        'TotalDrama': 0.8,
        'Area': 0.9
    };
    
    let budget = Number(input.shift());
    let count = Number(input.shift());

    for (let serial = 1; serial <= count; serial += 1) {
        let name = input.shift();
        let price = Number(input.shift());
        if (name in discounted) {price *= discounted[name];}
        budget -= price;
    }

    if (budget >= 0) {
        console.log(`You bought all the series and left with ${budget.toFixed(2)} lv.`);
    } else {
        console.log(`You need ${Math.abs(budget).toFixed(2)} lv. more to buy the series!`);
    }
}

series([10, 3, 'Thrones', 5, 'Riverdale', 5, 'Gotham', 2]);  // Should return:
// You bought all the series and left with 0.50 lv.

series([25, 6, 'Teen Wolf', 8, 'Protector', 5, 'TotalDrama', 5, 'Area', 4, 'Thrones', 5, 'Lucifer', 9]);  
// Should return:
// You need 2.00 lv. more to buy the series!